statsd/statsd.go

```diff
@@ -291,28 +320,38 @@ func (c *Client) flushLocked() error {
 	}
 
 	// clear the slice with a slice op, doesn't realloc
-	if cmdsFlushed == len(c.commands) {
-		c.commands = c.commands[:0]
+	if cmdsFlushed == len(b.commands) {
+		b.commands = b.commands[:0]
 	} else {
 		//this case will cause a future realloc...
 		// drop problematic command though (sorry).
-		c.commands = c.commands[cmdsFlushed+1:]
+		b.commands = b.commands[cmdsFlushed+1:]
```

Comment:
If we already checked that b.commands[cmdsFlushed] will fit in a UDP packet before we added it, why do we need to drop it at this point, just because we filled the packet?  Seems like this should be `b.commands = b.commands[cmdsFlushed:]`